cmd/goal/clerk.go

```diff
@@ -97,6 +98,69 @@ var clerkCmd = &cobra.Command{
 	},
 }
 
+func waitForCommit(client libgoal.Client, txid string) {
+	if noWaitAfterSend {
+		return
+	}
+
+	// Get current round information
+	stat, err := client.Status()
+	if err != nil {
+		reportErrorf(errorRequestFail, err)
+	}
+
+	for {
+		// Check if we know about the transaction yet
+		txn, err := client.PendingTransactionInformation(txid)
+		if err != nil {
+			reportErrorf(errorRequestFail, err)
+		}
+
+		if txn.ConfirmedRound > 0 {
+			reportInfof(infoTxCommitted, txid, txn.ConfirmedRound)
+			break
+		}
+
+		if txn.PoolError != "" {
+			reportErrorf(txPoolError, txid, txn.PoolError)
+		}
+
+		reportInfof(infoTxPending, txid, stat.LastRound)
+		stat, err = client.WaitForRound(stat.LastRound + 1)
+		if err != nil {
+			reportErrorf(errorRequestFail, err)
+		}
+	}
+}
+
+func writeTxnToFile(client libgoal.Client, dataDir string, walletName string, tx transactions.Transaction, filename string) {
+	var err error
+	var stxn transactions.SignedTxn
+	if sign {
```

Comment:
helper functions like this are easier to follow if you pass in the globals.  can you pass 'sign' in as an arg?